fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/UnnestedRecordTypeTest.java

```diff
@@ -1097,6 +1098,72 @@ void loadDoubleNestedType() {
         }
     }
 
+    //
+    // Loading failure tests
+    //
+    // Tests that assert on what happens when loadSyntheticRecord fails
+    //
+
+    @ParameterizedTest(name = "loadMapType[{index}]")
+    @MethodSource("mapMetaDataSuppliers")
+    void loadRecordNotFound(Function<RecordMetaDataHook, RecordMetaData> metaDataSource) {
+        final RecordMetaData metaData = metaDataSource.apply(addMapType());
+        final RecordType unnestedType = metaData.getSyntheticRecordType(UNNESTED_MAP);
+
+        try (FDBRecordContext context = openContext()) {
+            createOrOpenRecordStore(context, metaData);
+
+            for (TestRecordsNestedMapProto.OuterRecord outerRecord : sampleMapRecords()) {
```

Comment:
I don't think there's any reason to iterate over `sampleMapRecords()` here, none of them would save with a primary key of 100.

Reply:
Yes, of course. That was unnecessary.